exporter.go

```diff
@@ -62,11 +63,44 @@ func getEnvDuration(key string, defaultValue time.Duration) time.Duration {
 	return d
 }
 
-func createClientWithRetries(getClient func() (interface{}, error), retries int, retryInterval time.Duration) (interface{}, error) {
+// positiveDuration is a wrapper of time.Duration to ensure only positive values are accepted
+type positiveDuration struct{ duration *time.Duration }
+
+func (pd positiveDuration) String() string {
+	return fmt.Sprint(pd.duration)
+}
+
+func (pd positiveDuration) Set(s string) error {
+	if pd.isNegative(s) {
+		return fmt.Errorf("Negative duration is not valid")
+	}
+	dur, err := time.ParseDuration(s)
+	if err != nil {
+		return err
+	}
+
+	*pd.duration = dur
+	return nil
+}
+
+func (pd positiveDuration) isNegative(s string) bool {
+	return s[0] == '-'
+}
+
+func createPositiveDurationFlag(dur time.Duration, key, helper string) error {
```

Comment:
return a pointer to the positiveDuration, similarly to `flag.Uint`
```suggestion
func createPositiveDurationFlag(dur time.Duration, key, helper string)  *positiveDuration {
```